src/tracer/tracer.ts

```diff
@@ -88,6 +140,31 @@ export class AutoblocksTracer {
       args?.promptTracking ? { promptTracking: args.promptTracking } : {},
     );
 
+    if (args?.evaluators) {
+      try {
+        // Build semaphore registry
+        args.evaluators.forEach((evaluator) => {
```

Comment:
this is creating a semaphore every time someone calls sendEvent. but the evaluator concurrency setting is global; a user sets it because they don't want the given evaluator to be running more than `n` times concurrently _anywhere in their application_. this needs to be globally managed. you can follow the python sdk for reference:

* https://github.com/autoblocksai/python-sdk/blob/38d5ab5b294f95477500d4f1dd5702099ecd54ee/autoblocks/_impl/tracer.py#L29
* https://github.com/autoblocksai/python-sdk/blob/38d5ab5b294f95477500d4f1dd5702099ecd54ee/autoblocks/_impl/tracer.py#L130-L131